xpdan/glbl.py

```diff
@@ -101,3 +101,98 @@ class Glbl:
 
 
 an_glbl = Glbl()
+
+
+def make_glbl():
+    HOME_DIR_NAME = 'xpdUser'
+    BLCONFIG_DIR_NAME = 'xpdConfig'
+    BEAMLINE_HOST_NAME = 'xf28id1-ws2'
+    ARCHIVE_BASE_DIR_NAME = 'pe2_data/.userBeamtimeArchive'
+    USER_BACKUP_DIR_NAME = strftime('%Y')
+    OWNER = 'xf28id1'
+    BEAMLINE_ID = 'xpd'
+    GROUP = 'XPD'
+    DET_IMAGE_FIELD = 'pe1_image'
+    DARK_FIELD_KEY = 'sc_dk_field_uid'
+    CALIB_CONFIG_NAME = 'pyFAI_calib.yml'
+
+    # change this to be handled by an environment variable later
+    print(os.environ["IS_TEST"])
+    if int(os.environ["IS_TEST"]) == 1:
+        BASE_DIR = tempfile.mkdtemp()
+    else:
+        # We should load this from a user made config file
+        BASE_DIR = os.path.expanduser('~/')
+
+    # top directories
+    HOME_DIR = os.path.join(BASE_DIR, HOME_DIR_NAME)
+    BLCONFIG_DIR = os.path.join(BASE_DIR, BLCONFIG_DIR_NAME)
+    ARCHIVE_BASE_DIR = os.path.join(BASE_DIR, ARCHIVE_BASE_DIR_NAME)
+
+    # aquire object directories
+    CONFIG_BASE = os.path.join(HOME_DIR, 'config_base')
+    if int(os.environ["IS_TEST"]) == 1:
+        a = os.path.dirname(os.path.abspath(__file__))
+        b = a.split('glbl.py')[0]
+        os.makedirs(CONFIG_BASE)
+        shutil.copyfile(os.path.join(b, 'tests/pyFAI_calib.yml'),
+                        os.path.join(CONFIG_BASE, 'pyFAI_calib.yml'))
+    YAML_DIR = os.path.join(HOME_DIR, 'config_base', 'yml')
+    BT_DIR = YAML_DIR
+    SAMPLE_DIR = os.path.join(YAML_DIR, 'samples')
+    EXPERIMENT_DIR = os.path.join(YAML_DIR, 'experiments')
+    SCANPLAN_DIR = os.path.join(YAML_DIR, 'scanplans')
+    # other dirs
+    IMPORT_DIR = os.path.join(HOME_DIR, 'Import')
+    ANALYSIS_DIR = os.path.join(HOME_DIR, 'userAnalysis')
+    USERSCRIPT_DIR = os.path.join(HOME_DIR, 'userScripts')
+    TIFF_BASE = os.path.join(HOME_DIR, 'tiff_base')
+    USER_BACKUP_DIR = os.path.join(ARCHIVE_BASE_DIR, USER_BACKUP_DIR_NAME)
+
+    ALL_FOLDERS = [
+        HOME_DIR,
+        BLCONFIG_DIR,
+        YAML_DIR,
+        CONFIG_BASE,
+        SAMPLE_DIR,
+        EXPERIMENT_DIR,
+        SCANPLAN_DIR,
+        TIFF_BASE,
+        USERSCRIPT_DIR,
+        IMPORT_DIR,
+        ANALYSIS_DIR
+    ]
+
+    if int(os.environ["IS_TEST"]) == 1:
+        for folder in ALL_FOLDERS:
+            os.makedirs(folder, exist_ok=True)
+
+    # directories that won't be tar in the end of beamtime
+    _EXCLUDE_DIR = [HOME_DIR, BLCONFIG_DIR, YAML_DIR]
+    _EXPORT_TAR_DIR = [CONFIG_BASE, USERSCRIPT_DIR]
+
+    class Glbl:
+        beamline_host_name = BEAMLINE_HOST_NAME
```

Comment:
Yes, making this as a function means that we can produce "new" global every time. Otherwise if we just import a new copy of `glbl` then python is too cleaver and doesn't give us a new instance.